src/test/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSourceTest.java

```diff
@@ -0,0 +1,62 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import com.cloudbees.jenkins.GitHubWebHook;
+import hudson.util.LogTaskListener;
+import jenkins.scm.api.SCMHeadObserver;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+import com.github.tomakehurst.wiremock.junit.WireMockClassRule;
+import org.junit.runner.RunWith;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest( { GitHubWebHook.class })
+@PowerMockIgnore("javax.net.ssl.*")
+public class GitHubSCMSourceTest {
+    @ClassRule
+    public static WireMockClassRule wireMockRule = new WireMockClassRule(8089);
+    @Rule
+    public WireMockClassRule instanceRule = wireMockRule;
+
+    @Rule
+    public JenkinsRule jrule = new JenkinsRule();
+
+
+    @Test
+    public void testRetrieve(){
+        try {
+            mockStatic(GitHubWebHook.class);
+            when(GitHubWebHook.getJenkinsInstance()).thenReturn(jrule.getInstance());
```

Comment:
Is this really necessary? `Jenkins.getInstance()` will work from inside a test using `JenkinsRule` already.